src/internal_modules/roc_core/list.h

```diff
@@ -208,6 +208,19 @@ class List : public NonCopyable<> {
         insert_(element, &before);
     }
 
+    //! Insert element into list.
+    //!
+    //! @remarks
+    //!  - inserts @p element after @p after
+    //!  - acquires ownership of @p element
+    //!
+    //! @pre
+    //!  @p element should not be member of any list.
+    //!  @p after should be member of this list.
+    void insert_after(T& element, T& after) {
+        insert_(element, nextof(after));
+    }
```

Comment:
We can't use nextof() here. nextof() returns Pointer; if OwnershipPolicy is RefCountedOwnership, then Pointer will resolve to `SharedPtr<T>` instead of `T*`, and this code won't compile.

We could use `&*nextof(after)`, but this would give unnecessary overhead for creating/destroying SharedPtr. I guess instead it would be better to obtain that pointer manually.

Actually, I think it makes sense to change `insert_()` second argument from `T* before` to `ListNode::ListNodeData* before`. For `insert_before()` and `push_back()` there is no difference, but for `insert_after()` we'll save unnecessary conversion: from `ListNodeData*` to `T*` and then back to `ListNodeData*`. Instead we can pass `ListNodeData*` immediately.

Reply:
Thanks for the clarification! This indeed makes sense. 

I adjusted the code according to your last suggestion.
Contrary to what you write (at least this is how I understood you), changes must be made to the all corresponding insert functions.
The changes are rather minor, though.

Please let me know what you think!

Reply:
Yes, sorry, I mean that their *behavior* is not affected. However there are syntactic changes, yes.